include/sway/config.h

```diff
@@ -110,12 +110,19 @@ struct bar_config {
 	 * Always visible in dock mode. Visible only when modifier key is held in hide mode.
 	 * Never visible in invisible mode.
 	 */
-	char *mode;
+	char *display_mode;
 	/**
 	 * One of "show" or "hide".
 	 *
 	 * In "show" mode, it will always be shown on top of the active workspace.
 	 */
+	char *mode;
+	/**
+	 * Current keybinding mode
+	 *
+	 * Names should probably be reworked at some point, but I'm keeping it for now
+	 * Don't think this does anything at the moment
+	 */
 	char *hidden_state;
```

Comment:
`hidden_state` is naturally a bool value, and as its a state variable rather than a storing config option value maybe it's better to change its type? Also avoid redundant strcmp in the code.